conans/client/graph/graph_builder.py

```diff
@@ -184,17 +190,23 @@ def _expand_require(self, require, node, graph, check_updates, update,
                                  % (node.ref, require.ref))
 
         # If the requirement is found in the node public dependencies, it is a diamond
-        previous = node.public_deps.get(name)
-        previous_closure = node.public_closure.get(name)
+        previous = node.public_deps.get(name, context=context)
+        previous_closure = node.public_closure.get(name, context=context)
         # build_requires and private will create a new node if it is not in the current closure
         if not previous or ((require.build_require or require.private) and not previous_closure):
             # new node, must be added and expanded (node -> new_node)
+            if context == CONTEXT_BUILD and not profile_build:
+                raise ConanException("Package '{}' requires '{}' on context build, but no profile"
+                                     " for build_machine has been provided. See docs about"
+                                     " crossbuilding feature.".format(node, name))
```

Comment:
We need to provide a link here

Reply:
You mean a link to some page in the docs?

Reply:
Yes, seems unreachable, lets remove it, done.